src/parquet_impl.cpp

```diff
@@ -331,8 +331,11 @@ convert_const(Const *c, Oid dst_oid)
                 getTypeOutputInfo(c->consttype, &output_fn, &isvarlena);
                 getTypeInputInfo(dst_oid, &input_fn, &input_param);
 
-                str = DatumGetCString(OidOutputFunctionCall(output_fn,
-                                                            c->constvalue));
+                // pg_16 update
+                char* output = OidOutputFunctionCall(output_fn, c->constvalue);
+                Datum datumValue = PointerGetDatum(output);
+                str = DatumGetCString(datumValue);
+                // str = DatumGetCString(OidOutputFunctionCall(output_fn, c->constvalue));
```

Comment:
Looks like it isn't necessary to call `DatumGetCString()` in the first place. You can remove it at all and do the following:

```suggestion
                str = OidOutputFunctionCall(output_fn, c->constvalue);
```

This is possible because `OidOutputFunctionCall()` already converts Datum to C string:
https://github.com/postgres/postgres/blob/REL_16_3/src/backend/utils/fmgr/fmgr.c#L1739
https://github.com/postgres/postgres/blob/REL_16_3/src/backend/utils/fmgr/fmgr.c#L1656